spring-xd-dirt/src/main/java/org/springframework/xd/dirt/module/store/ZooKeeperComposedModuleDefinitionRegistry.java

```diff
@@ -163,9 +169,10 @@ public ModuleDefinition findDefinition(String name, ModuleType type) {
 						Paths.build(Paths.MODULES, type.toString(), child));
 				// Check for data (only composed modules have definitions)
 				if (data != null && data.length > 0) {
-
 					ModuleDefinition composed = this.findDefinition(child, type);
-					results.add(composed);
+					if (composed != null) {
```

Comment:
We shouldn't ignore if the composed module is `null` here. In such case, we should fail fast. We can throw `NoSuchModuleException` here.